### PR TITLE
Generate C/Fortran API docs with Doxygen

### PR DESCRIPTION
**Pull request by sloede**
To generate the docs, go to `docs/doxygen` and run `doxygen` without any arguments. You can then view the result by opening the file `docs/doxygen/build/html/index.html` in your browswer.

This is just a very early draft after playing around some (too much) time with Doxygen 🙄 One particular challenge is the fact that we have to document APIs in two languages (actually three, if you include Julia), and that some function names are identical. I believe that this similarity is the reason why, e.g., the documentation for the Fortran interface might not show up properly (or maybe it just doesn't support Fortran interfaces).

Instead of using Documenter.jl for the main docs and Doxygen for the C/Fortran API, we could also consider to use Doxygen only for the C part and using something like [FORD](https://github.com/Fortran-FOSS-Programmers/ford) for the Fortran part.

Anyhow, this is as far as I'm going to take it with docs. @bgeihe feel free to adopt this PR, add to it, or just close it again if you don't like it - I just didn't want to get the time I put into this go to waste entirely :-)

The following things one would have to do if willing to make this PR work:
- [x] Figure out how to properly document the Fortran API
- [x] Auto-generate the docs as part of the call to Documenter.jl and put the result in a subdirectory
- [x] Link to C/Fortran API from main docs

### CODE REVIEW

**Comment by sloede** [comment]
Preview the docs here: https://trixi-framework.github.io/libtrixi/previews/PR52

Go to "Reference" ➡️ "C/Fortran", where you will find a link to the doxygen-generated docs: https://trixi-framework.github.io/libtrixi/c-fortran-api

File: .github/workflows/Documenter.yml
Comment (sloede) [inline .github/workflows/Documenter.yml:63]:
```suggestion
        if: ${{ github.ref == 'refs/heads/main' }}
```
Needs to be applied before merging the PR, otherwise future PR builds will overwrite the main docs

File: docs/src/reference-julia.md
Comment (bgeihe) [inline docs/src/reference-julia.md:10]: This refers to users, who want to use Trixi but need not use the C / Fortran interface?

**Comment by bgeihe** [comment]
I would like to check the output of [moxygen](https://github.com/sourcey/moxygen), but I can do this in a separate as well.

File: utils/libtrixi-init-julia
Comment (bgeihe) [inline utils/libtrixi-init-julia:323]:
```suggestion
echo "will then automatically be set. It may be overridden manually."
```

File: docs/src/reference-julia.md
Comment (sloede) [reply docs/src/reference-julia.md:10]: Yes exactly. Do you think it is unclear? Feel free to change/improve it 👍

File: .github/workflows/Documenter.yml
Comment (sloede) [reply .github/workflows/Documenter.yml]: Not needed anymore.

**Comment by sloede** [comment]
> I would like to check the output of [moxygen](https://github.com/sourcey/moxygen), but I can do this in a separate PR as well.

The current state of the PR is that everything _seems_ to work as it should, including the placement of the C/Fortran API into the proper folder such that it respects the versioning of Documenter.jl.

I thus recommend we go ahead and merge this PR in its current state (after a review by you, of course). All improvements can then be iterated on in a new PR.

File: docs/src/reference-julia.md
Comment (bgeihe) [reply docs/src/reference-julia.md:10]: Thinking about it again, it seems clear.

**Review by bgeihe — APPROVED** [review]

**Pull request merged**
